### PR TITLE
[minor] Move More Modules to `utils` Sub-Package

### PR DESCRIPTION
**Pull request by ric-evans**
closes https://github.com/WIPACrepo/rest-tools/issues/53

Note: not backwards compatible

moves
``` python
server.auth.Auth # along with alias: server.Auth
server.auth.OpenIDAuth # along with alias: server.OpenIDAuth
server.config.from_environment # along with alias: server.from_environment
server.daemon.Daemon # along with alias: server.Daemon
```
to
`utils.*`

### CODE REVIEW

**Comment by ric-evans** [comment]
This will break some imports from clients. Seeing that no functionality has changed and only import paths need to change, it's worth it.

**Comment by ric-evans** [comment]
This will also avoid activating tracing when only utility functionality is wanted.

File: rest_tools/server/__init__.py [hunk 1]
```diff
@@ -21,8 +19,4 @@
     "catch_error",
     "role_authorization",
     "scope_role_auth",
-    "Auth",
-    "OpenIDAuth",
-    "Daemon",
-    "from_environment",
 ]
```
Comment (dsschult) [inline rest_tools/server/__init__.py:22]: I guess the question is, do we want to be backwards compatible and do 
```python
from ..utils.auth import Auth, OpenIDAuth
```
or should we make a minor (major?) release and break things?  If so, we could probably break a few more things. :smile:

Comment (ric-evans) [reply rest_tools/server/__init__.py:22]: As it stands currently, I'm in favor of making this a minor release. Will it break things? Yes. Does it alter actual functionality? No.

Comment (ric-evans) [reply rest_tools/server/__init__.py:22]: >If so, we could probably break a few more things. :smile:

What are you looking at adding?

Comment (dsschult) [reply rest_tools/server/__init__.py:22]: Not sure.  A minor release I'd probably let go as-is, whereas a major release I'd be tempted to do more.

Comment (ric-evans) [reply rest_tools/server/__init__.py:22]: let's keep it minor then

**Pull request merged**